yt_dlp/extractor/neteasemusic.py

```diff
@@ -40,12 +45,20 @@ def _encrypt(cls, dfsid):
         result = b64encode(m.digest()).decode('ascii')
         return result.replace('/', '_').replace('+', '-')
 
-    def make_player_api_request_data_and_headers(self, song_id, bitrate):
+    def _create_eapi_cipher(self, api_path, query, cookies):
         KEY = b'e82ckenh8dichen8'
-        URL = '/api/song/enhance/player/url'
-        now = int(time.time() * 1000)
-        rand = randint(0, 1000)
-        cookie = {
+        request_text = json.dumps({**query, 'header': cookies}, separators=(',', ':'))
+
+        message = f'nobody{api_path}use{request_text}md5forencrypt'.encode('latin1')
+        msg_digest = md5(message).hexdigest()
+
+        data = pkcs7_padding(bytes_to_intlist(
+            f'{api_path}-36cd479b6b5-{request_text}-36cd479b6b5-{msg_digest}'))
+        encrypted = intlist_to_bytes(aes_ecb_encrypt(data, bytes_to_intlist(KEY)))
+        return b'params=' + hexlify(encrypted).upper()
```

Comment:
As a sidenote, we can use builtins (`bytes_to_intlist` => `list`, `intlist_to_bytes` => `bytes`, `hexlify(b'')` => `b''.hex().encode()`) for conversion:
```py
        data = pkcs7_padding(list(b'-36cd479b6b5-'.join(map(str.encode, [
            api_path, request_text, msg_digest]))))
        encrypted = bytes(aes_ecb_encrypt(data, list(KEY)))
        return f'params={encrypted.hex().upper()}'.encode()
```
Not necessary changes, but use if you like any

Reply:
will take a look later